### PR TITLE
Feature: Time series

### PR DESCRIPTION
**Pull request by samuelburbulla**
# Feature: Time series

### Which issue does this PR tackle?

  - Add an example that demonstrates a discretization-invariant operator. 

### How does it solve the problem?

  - Implements an example on time series forecasting with BelNet.

### How are the changes tested?

  - Added timeseries notebook.

## Checklist for Contributors

- [x] Scope: This PR tackles exactly one problem.
- [x] Conventions: The branch follows the `feature/title-slug` convention.
- [x] Conventions: The PR title follows the `Bugfix: Title` convention.
- [x] Coding style: The code passes all pre-commit hooks.
- [x] Documentation: All changes are well-documented.
- [x] Tests: New features are tested and all tests pass successfully.
- [x] Changelog: Updated CHANGELOG.md for new features or breaking changes.
- [x] Review: A suitable reviewer has been assigned.

## Checklist for Reviewers:

- [x] The PR solves the issue it claims to solve and only this one.
- [x] Changes are tested sufficiently and all tests pass.
- [x] Documentation is complete and well-written.
- [x] Changelog has been updated, if necessary.

### CODE REVIEW

File: examples/timeseries.ipynb [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+    "plt.rcParams[\"axes.facecolor\"] = (1, 1, 1, 0)\n",
+    "plt.rcParams[\"figure.facecolor\"] = (1, 1, 1, 0)\n",
+    "plt.rcParams[\"legend.framealpha\"] = 0.0"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problem\n",
```
Comment (JakobEliasWagner) [inline examples/timeseries.ipynb:109]: I can see that the problem itself is not the focus of this notebook. But I feel like it would be beneficial to give a slightly longer introduction to the problem itself. Furthermore it could be nice to emphasize the temporal aspect of this problem ("sequence of observations" instead of "time grid"). Additionally are both intervals implemented in the code section in a right open fashion (otherwise 1 would be in the future and history simultaneously).

Comment (JakobEliasWagner) [reply examples/timeseries.ipynb:109]: Adapted suggestion:
```
## Problem

As a simple demonstration of the concepts, we will consider for sequences of $n$ observations $f_i$, with $i\in[1, n]$. The index $i$ corresponds to samples of a function $f$ at times $t_i$. For these observations of the function we would like to predict $m-n$ future values $f_j$, where $j\in[n+1, m]$. In this example we choose a simple sine function
$$
f(t) = \sin(2\pi t),
$$ 
and choose $t_i\in [0, 1)$, $n=32$, $t_j\in [1, 1.5)$, and $m-n = 16$. Both $t_i$ and $t_j$ are sampled uniformly from their intervals. The trained neural operator is supposed to predict the future of the time series, by only having access to the historical samples.
```

File: examples/timeseries.ipynb [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plt.plot(t_hist, f_hist, \"g.\", label=r\"$f_{hist}$\")\n",
```
Comment (JakobEliasWagner) [inline examples/timeseries.ipynb:170]: For alignment with the other notebooks and accuracy in formulation these plots should be $f_{hist}(t)$ and  $f_{fut}(t)$.

File: examples/timeseries.ipynb [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "# Plot inputs and predictions\n",
+    "plt.plot(x.squeeze(), u.squeeze(), \"g.\", label=r\"$f_{hist}$\")\n",
```
Comment (JakobEliasWagner) [inline examples/timeseries.ipynb:322]: Here also $f_{hist}(t)$ ...

File: examples/timeseries.ipynb [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+    "Trainer(operator).fit(dataset, tol=1e-5)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Evaluation\n",
+    "\n",
+    "Now, let's plot the predictions of the trained BelNet on the interval [1, 1.5].\n",
```
Comment (JakobEliasWagner) [inline examples/timeseries.ipynb:266]: Mathematical descriptions should be in equation format. Here the interval is again the right-open interval $[1, 1.5)$.

File: examples/timeseries.ipynb [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+    "\n",
+    "Remarkably, the operator makes a good prediction even if we sample the sine wave\n",
```
Comment (JakobEliasWagner) [inline examples/timeseries.ipynb:268]: In this notebook the term "discretization-invariant" is used to introduce and (somewhat) motivate this example. In the formulation from the [BelNet](https://arxiv.org/pdf/2212.07336.pdf) paper this term only applies to the discretization of the input domain. 

In this evaluation two concepts are mixed (discretization invariance and domain independence). I think more clarity on the purpose of each plot would be nice. The last sentence of the md section covers the domain independence, and a second sentence explicitly mentioning the discretization invariance could clear up this confusion.

File: examples/timeseries.ipynb (hunk 2 above)
Comment (samuelburbulla) [reply examples/timeseries.ipynb:170]: In this context, f_hist and f_fut are tensors, not functions, so f_hist(t) doesn't make sense here. We only plot the value in f_hist as defined in the code.

File: examples/timeseries.ipynb (hunk 5 above)
Comment (samuelburbulla) [reply examples/timeseries.ipynb:268]: makes sense to mention it!

File: examples/timeseries.ipynb (hunk 3 above)
Comment (samuelburbulla) [reply examples/timeseries.ipynb:322]: Same as above, but you're right we should write $f_pred(t)$
